src/linker/pipeline.py

```diff
@@ -2,24 +2,49 @@
 from pathlib import Path
 from typing import Callable, Dict, Optional, Tuple
 
-from loguru import logger
-
 from linker.configuration import Config
 from linker.implementation import Implementation
-from linker.pipeline_schema import PIPELINE_SCHEMAS, PipelineSchema
-from linker.step import Step
+from linker.utilities.general_utils import exit_with_validation_error
 
 
 class Pipeline:
     """Abstraction to handle pipeline specification and execution."""
 
     def __init__(self, config: Config):
         self.config = config
-        self.steps = self._get_steps()
         self.implementations = self._get_implementations()
-        self._validation_errors = {}
         self._validate()
 
+    def _get_implementations(self) -> Tuple[Implementation, ...]:
```

Comment:
Tiny nit: move these setup/init methods to the bottom (which is how the other classes are structured)

Reply:
done